### PR TITLE
Stardew Valley: Cut tests by 3 minutes

### PR DESCRIPTION
**Pull request by black-sliver**
Reduce test duplication.

Also switch some `SVTestBase` to a new `SVTestCase` where we don't even use the default-constructed test world. Tests need further cleanup, but it's a start.

My main concern is that stardew tests all the mods and all the ER and all the items in individual (slowly) generated complete solo seeds, which is not necessarily the best approach to unit testing if time is already invested in writing tests. (In contrast, the generic tests can only ever work on fully constructed seeds, because there is no extra information about games available to core).

### CODE REVIEW

**Review by agilbert1412 — APPROVED** [review]
I have discussed in Discord, and would prefer using subtests on the tests that were merged to retain the general convenience of the results format.

**Comment by black-sliver** [comment]
I pushed 3 more commits, 2 of them regarding the ideas in Discord, so whenever you have the time to check if that's good enough @agilbert1412 , we can probably merge it.

I didn't find other good places to use the `skip_extra_tests`, but I believe some stuff in TestGeneration.py could also be optional without losing a lot of coverage.

File: worlds/stardew_valley/test/TestBackpack.py [hunk 1]
```diff
@@ -39,14 +39,7 @@ def run_default_tests(self) -> bool:
         return False
 
     def test_backpack(self):
-        # 2 items
-        item_names = [item.name for item in self.multiworld.get_items()]
-        self.assertEqual(item_names.count("Progressive Backpack"), 2)
+        super().test_backpack()
```
Comment (agilbert1412) [inline worlds/stardew_valley/test/TestBackpack.py:42]: That's a way to use inheritance that I hadn't thought of

**Review by agilbert1412 — APPROVED** [review]
Other candidates can probably be found for skipping extra tests. If this is desired, I would recommend taking a look at friendsanity tests first, based on previous experiences. Other high-check-count options that have quite a bit of somewhat-irregular logic would be good candidates as well, notably museumsanity settings.

At any rate, this PR looks fine as-is

**Pull request merged**